runtime/runtime/src/state_viewer/mod.rs

```diff
@@ -152,6 +152,111 @@ impl TrieViewer {
         access_keys
     }
 
+    pub fn view_gas_key(
+        &self,
+        state_update: &TrieUpdate,
+        account_id: &AccountId,
+        public_key: &PublicKey,
+        include_nonces: bool,
+    ) -> Result<near_primitives::views::GasKeyView, errors::ViewGasKeyError> {
+        let gas_key = get_gas_key(state_update, account_id, public_key)?.ok_or_else(|| {
+            errors::ViewGasKeyError::GasKeyDoesNotExist { public_key: public_key.clone() }
+        })?;
+        let nonces = if include_nonces {
+            let mut nonces = Vec::new();
+            for nonce_index in 0..gas_key.num_nonces {
+                let nonce = get_gas_key_nonce(state_update, account_id, public_key, nonce_index)?
+                    .ok_or_else(|| errors::ViewGasKeyError::InternalError {
+                    error_message: format!(
+                        "Unexpected missing nonce for gas key {:?} at index {}",
+                        public_key, nonce_index
+                    ),
+                })?;
+                nonces.push(nonce);
+            }
+            Some(nonces)
+        } else {
+            None
+        };
+        Ok(near_primitives::views::GasKeyView {
+            num_nonces: gas_key.num_nonces,
+            balance: gas_key.balance,
+            permission: gas_key.permission.into(),
+            nonces,
+        })
+    }
+
+    pub fn view_gas_keys(
+        &self,
+        state_update: &TrieUpdate,
+        account_id: &AccountId,
+    ) -> Result<Vec<near_primitives::views::GasKeyInfoView>, errors::ViewGasKeyError> {
+        let prefix = trie_key_parsers::get_raw_prefix_for_gas_keys(account_id);
+        let mut result: Vec<near_primitives::views::GasKeyInfoView> = Vec::new();
+        for raw_key in state_update.iter(&prefix)? {
+            let raw_key = raw_key?;
+            let trie_key = parse_trie_key_gas_key_from_raw_key(&raw_key).map_err(|err| {
+                errors::ViewGasKeyError::InternalError {
+                    error_message: format!("Failed to parse gas key from raw key: {:?}", err),
+                }
+            })?;
+            let near_primitives::trie_key::TrieKey::GasKey { public_key, index, .. } = &trie_key
+            else {
+                return Err(errors::ViewGasKeyError::InternalError {
+                    error_message: "Unexpected trie key type for gas key".to_string(),
+                });
```

Comment:
Would we want to log or include in the error the unexpected trie key and some additional info here to make it easier to debug if this is encountered?